agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -169,310 +95,328 @@ public CallReplacementTransformer() {}
       byte[] classfileBuffer)
       throws IllegalClassFormatException {
 
-    // debug = className.equals ("chicory/Test");
+    if (className == null) {
+      return null;
+    }
 
     String fullClassName = className.replace("/", ".");
 
-    debug_transform.log("In Transform: class = %s%n", className);
-
-    // Don't instrument boot classes. We only want to instrument
-    // user classes classpath.
-    // Most boot classes have the null loader,
-    // but some generated classes (such as those in sun.reflect) will
-    // have a non-null loader. Some of these have a null parent loader,
-    // but some do not. The check for the sun.reflect package is a hack
-    // to catch all of these. A more consistent mechanism to determine
-    // boot classes would be preferrable.
-    if (loader == null) {
-      debug_transform.log("ignoring system class %s, class loader == null", fullClassName);
-      return (null);
-    } else if (loader.getParent() == null) {
-      debug_transform.log("ignoring system class %s, parent loader == null\n", fullClassName);
-      return (null);
-    } else if (fullClassName.startsWith("sun.reflect")) {
-      debug_transform.log("ignoring system class %s, in sun.reflect package", fullClassName);
-      return (null);
-    } else if (fullClassName.startsWith("com.sun")) {
-      System.out.printf("Class from com.sun package %s with nonnull loaders\n", fullClassName);
+    if (isNonGUIBootClass(loader, fullClassName)) {
```

Comment:
How about moving this method call into `isExcludedClass`?  That is, clients such as this method can call only `isExcludedClass` rather than having to remember to also call this method.